File: src/main/kotlin/com/wynnlab/spells/assassin/Multihit.kt
package com.wynnlab.spells.assassin

import com.wynnlab.api.normalizeOnXZ
import com.wynnlab.spells.SpellL
import com.wynnlab.spells.SpellData
import com.wynnlab.util.plus
import org.bukkit.Location
import org.bukkit.Sound
import org.bukkit.entity.Entity
import org.bukkit.entity.Mob
import org.bukkit.entity.Player

class Multihit(player: Player) : SpellL(player, 11, SpellData.MULTIHIT) {
    private lateinit var l: Location
    private lateinit var entities: Collection<Entity>

    override fun tick() {
        when {
            tick == 0 -> {
                player.playSound(player.location, Sound.ENTITY_PLAYER_ATTACK_STRONG, .5f, 1f)
                player.playSound(player.location, Sound.ENTITY_IRON_GOLEM_HURT, 1f, 1.5f)
                player.playSound(player.location, Sound.ENTITY_BLAZE_SHOOT, 1f, 1.3f)
                if (clone) player.playSound(player.location, Sound.ENTITY_BLAZE_AMBIENT, .5f, 1.6f)

                l = player.location + player.eyeLocation.direction.clone().normalizeOnXZ()

                entities = player.world.getNearbyEntities(l, 2.0, 2.0, 2.0) {
                    it !is Player && it is Mob
                }
            }
            tick < 10 -> {
                for (e in entities) {

                }
            }
            else -> {

            }
        }
    }
}

File: src/main/resources/classes/assassin/multihit.py
from org.bukkit import Particle, Sound
from org.bukkit.potion import PotionEffectType

from com.wynnlab.spells import PySpell
from com.wynnlab.util import BukkitUtils

class Spell(PySpell):
    def __init__(self):
        self.l = None
        self.entities = None
        self.shift = False

    def init(self):
        self.shift = self.player.isSneaking()

    def tick(self):
        if self.t % 2 != 0:
            return

        if self.t == 0:
            if self.player.hasPotionEffect(PotionEffectType.INVISIBILITY):
                self.castSpell('ASSASSIN', 5)

            self.sound(Sound.ENTITY_PLAYER_ATTACK_STRONG, .5, 1)
            self.sound(Sound.ENTITY_IRON_GOLEM_HURT, 1, 1.5)
            if self.clone:
                self.sound(Sound.ENTITY_BLAZE_AMBIENT, .3, 1.5)

            v = BukkitUtils.normalizeOnXZ(self.player.getEyeLocation().getDirection().clone())
            self.l = self.player.getLocation().clone().add(v).add(0, .5, 0)

            self.particle(self.l, Particle.SWEEP_ATTACK, 5, .5, .5, .5, .1)

            self.entities = self.nearbyMobs(self.l, 3, 3, 3)

            self.particle(self.l.add(v), Particle.SWEEP_ATTACK, 5, .5, .5, .5, .1)
            self.particle(self.l.add(v), Particle.SWEEP_ATTACK, 5, .5, .5, .5, .1)

        elif self.t <= 20:
            for e in self.entities:
                e.setVelocity(self.player.getEyeLocation().getDirection().clone().multiply(.05 if self.shift else .3).setY(.2).rotateAroundY((.1 * self.t) if self.t % 2 == 0 else (-.1 * self.t)))

                self.particle(e.getLocation(), Particle.SWEEP_ATTACK, 5, .5, .5, .5, .5)
                if self.clone:
                    self.particle(e.getLocation(), Particle.SPELL_WITCH, 7, .5, .5, .5, .2)
                self.particle(e.getLocation(), Particle.SQUID_INK, 6, .5, .5, .5, .1)
                self.particle(e.getLocation(), Particle.CRIT, 7 if self.clone else 10, .5, .5, .5, .1)

                self.sound(e.getLocation(), Sound.ENTITY_PLAYER_ATTACK_SWEEP, 1, 1.3)
                self.sound(e.getLocation(), Sound.ENTITY_PLAYER_ATTACK_CRIT, .8, 1.6)

                self.damage(e, 1)

        else:
            for e in self.entities:
                if not self.shift:
                    e.setVelocity(self.player.getEyeLocation().getDirection().clone().setY(.5))

                self.sound(e.getLocation(), Sound.ENTITY_PLAYER_ATTACK_KNOCKBACK, 1, 1.3)

                self.damage(e, 6)
                if self.clone:
                    self.sound(e.getLocation(), Sound.ENTITY_BLAZE_DEATH, 1, 1.2)
                    self.sound(e.getLocation(), Sound.ENTITY_BLAZE_AMBIENT, 1, 1.6)
                    self.sound(e.getLocation(), Sound.ENTITY_FIREWORK_ROCKET_BLAST_FAR, .5, 1)
